### PR TITLE
Support collection of custom stats in stats frontend module

### PR DESCRIPTION
**Pull request by zero-24**
#### Summary of Changes

Similar to how the backend module triggers an event to add additional items to its output, this adds support for the `mod_stats` module to do the same.

A new `onGetStats` event is published with a single parameter, a context (hardcoded to `mod_stats_admin`) and any plugin responding to this event is expected to return an array matching this schema:

``` php
array(
    'title' => 'Stats Item Title',
    'data' => 'Data to be displayed',
    'icon' => 'Optional IcoMoon font icon for the row, defaulting to "info" if not set',
);
```

The title and data attributes MUST be defined or a row is not added.

Similar to the quickicons implementation, items can NOT be removed via the plugin event.

#### Testing Instructions

Add the following method to a system plugin:

``` php
    public function onGetStats($context)
    {
        return array(
            array(
                'title' => 'Plays Audio',
                'data'  => JText::_('JNO'),
                'icon'  => 'music',
            ),
        );
    }
```

With the admin stats module enabled, you should see a new row with this data.

Based on the same thing that @mbabker did for the backend: https://github.com/joomla/joomla-cms/pull/10870

The weblinks part is moved here: https://github.com/joomla-extensions/weblinks/pull/333/files to be included in the 3.7 version.

### CODE REVIEW

**Comment by zero-24** [comment]
@franz-wohlkoenig as you have tested the weblinks PR you have also tested here ;)

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on 9883fad1e6594ad5bc5f95194a43c05ff3700945<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15138">issues.joomla.org/tracker/joomla-cms/15138</a>.</sub>

**Comment by sanderpotjer** [comment]
@zero-24 adding the onGetStats function to a system plugin already adding a new row before applying your patch?

**Comment by zero-24** [comment]
Yes in the backend but not in the frontend module.

**Comment by sanderpotjer** [comment]
I have tested this item :red_circle: unsuccessfully on 9883fad1e6594ad5bc5f95194a43c05ff3700945<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15138">issues.joomla.org/tracker/joomla-cms/15138</a>.</sub>

**Comment by sanderpotjer** [comment]
The custom data is displayed instead of the "article hits" when "Hit Counter" is set to yes in the module options.

Before patch:
screen shot 2017-05-01 at 09 51 43

After patch:
screen shot 2017-05-01 at 09 51 53

So the custom stats data is shown, but the hits value is gone.

**Comment by zero-24** [comment]
> The custom data is displayed instead of the "article hits" when "Hit Counter" is set to yes in the module options.

fixed please test again @sanderpotjer

**Comment by sanderpotjer** [comment]
I have tested this item :white_check_mark: successfully on f9dd52adfebaaa8fe2ae15b47d9ccb1624755f2d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15138">issues.joomla.org/tracker/joomla-cms/15138</a>.</sub>

**Comment by sanderpotjer** [comment]
@zero-24 thanks, that fixed it!

**Comment by astridx** [comment]
I have tested this item :white_check_mark: successfully on f9dd52adfebaaa8fe2ae15b47d9ccb1624755f2d<br /><br />I tested this successfully.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15138">issues.joomla.org/tracker/joomla-cms/15138</a>.</sub>

**Comment by astridx** [comment]
I have tested this item :white_check_mark: successfully on f9dd52adfebaaa8fe2ae15b47d9ccb1624755f2d<br /><br />I tested this successfully.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15138">issues.joomla.org/tracker/joomla-cms/15138</a>.</sub>

File: modules/mod_stats/helper.php [hunk 1]
```diff
@@ -167,6 +138,29 @@ public static function &getList(&$params)
...
 				$rows[$i]->title = JText::_('MOD_STATS_ARTICLES_VIEW_HITS');
 				$rows[$i]->data  = $hits + $increase;
+				$i++;
+			}
+		}
+
+		// Include additional data defined by published system plugins
+		JPluginHelper::importPlugin('system');
+
+		$app    = JFactory::getApplication();
```
Comment (zero-24) [inline modules/mod_stats/helper.php]: fixed

**Comment by gwsdesk** [comment]
I have tested this item :white_check_mark: successfully on 84c2ee70441745438c9fdf33e8c2755b5c925a21<br /><br />I have tested this item ✅ successfully on f9dd52a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15138">issues.joomla.org/tracker/joomla-cms/15138</a>.</sub>

**Comment by gwsdesk** [comment]
I have tested this item ✅ successfully on [f9dd52a](https://github.com/joomla/joomla-cms/commit/f9dd52adfebaaa8fe2ae15b47d9ccb1624755f2d "")<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/15138">issues.joomla.org/tracker/joomla-cms/15138</a>.</sub>

**Pull request merged**